### ISSUE TITLE
Cannot pattern match an array with literals and spread

### ISSUE THREAD

**Issue by cie** [mint-lang/mint#488]
This does not work:

```
module M {
  fun a(d : Array(Number)) : String {
    case (d) {
      [3, ...rest] => "ok"
      => "not ok"
    }
  }
}
```
Maybe it's not implemented, but the error message is misleading:
> I was looking for the name of the field of a record to access but found ..rest] instead.

**Comment by gdotdesign** [comment]
Yes, it's not implemented yet, will be part of #226

**Comment by cie** [comment]
Thanks!
